### PR TITLE
Clean up analytics

### PR DESCRIPTION
**Pull request by jeanp413**

### CODE REVIEW

**Comment by jeanp413** [comment]
@mustard-mh could you test this :pray:

**Comment by mustard-mh** [comment]
Just notice this PR, going to rebase and force push

File: src/vs/gitpod/browser/gitpodInsightsAppender.ts [hunk 1]
```diff
@@ -4,32 +4,69 @@
...
+		productName: string,
+		private readonly productVersion: string,
+		private readonly gitpodPreview?: IGitpodPreviewConfiguration,
+		readonly galleryServiceUrl?: string
 	) {
+		this._asyncAIClient = null;
 		this._baseProperties = {
-			appName: this.productService.nameShort,
+			appName: productName,
 			uiKind: 'web',
```
Comment (mustard-mh) [inline src/vs/gitpod/browser/gitpodInsightsAppender.ts:37]: Should we make version a common properties?

File: src/vs/gitpod/browser/gitpodInsightsAppender.ts [hunk 2]
```diff
@@ -40,42 +77,51 @@ export class GitpodInsightsAppender implements ITelemetryAppender {
...
-				if (this.productService.gitpodPreview?.log?.analytics) {
-					console.log('Gitpod Analytics: ', JSON.stringify(trackMessage, undefined, 2));
-				}
-			} else {
-				// TODO(ak) get rid of it
-				// it is bad usage of window.postMessage
-				// we should use Segment directly here and publish to production/staging untrusted
-				// use server api to resolve a user
-				window.postMessage({ type: 'vscode_telemetry', event: trackMessage.event, properties: trackMessage.properties }, '*');
+			if (process.env['VSCODE_DEV'] && this.gitpodPreview?.log?.analytics) {
```
Comment (mustard-mh) [inline src/vs/gitpod/browser/gitpodInsightsAppender.ts:86]: [nit] Should we always use `this.devMode` or `process.env['VSCODE_DEV']` and only keep one of them in the same file?

File: src/vs/gitpod/browser/gitpodInsightsAppender.ts [hunk 3]
```diff
@@ -40,42 +77,51 @@ export class GitpodInsightsAppender implements ITelemetryAppender {
...
+			return undefined;
+		}
+		const { workspaceId, instanceId, debugWorkspaceType, ownerId } = await this.getWorkspaceInfo();
+		mappedEvent.properties = filter(mappedEvent.properties, (_, v) => v !== undefined && v !== null);
+		mappedEvent.properties = {
+			...mappedEvent.properties,
+			...this._baseProperties,
+			workspaceId,
+			instanceId,
+			debugWorkspace: debugWorkspaceType !== 'noDebug',
```
Comment (mustard-mh) [inline src/vs/gitpod/browser/gitpodInsightsAppender.ts:110]:
```suggestion
			debugWorkspace: String(debugWorkspaceType !== 'noDebug'),
```

File: src/vs/gitpod/browser/gitpodInsightsAppender.ts [hunk 4]
```diff
@@ -86,101 +132,95 @@ export class GitpodInsightsAppender implements ITelemetryAppender {
...
+				});
+				if (!infoResponse.ok) {
+					throw new Error(`Getting workspace info failed: ${infoResponse.statusText}`);
+				}
+				const info: SupervisorWorkspaceInfo = await infoResponse.json();
+				return {
+					gitpodHost: this.devMode ? this.gitpodPreview?.host ?? 'gitpod-staging.com' : new URL(info.gitpodHost).host,
+					instanceId: info.instanceId,
+					workspaceId: info.workspaceId,
+					debugWorkspaceType: info.debugWorkspaceType || 'noDebug',
```
Comment (mustard-mh) [inline src/vs/gitpod/browser/gitpodInsightsAppender.ts:218]:
```suggestion
					debugWorkspaceType: String(info.debugWorkspaceType || 'noDebug'),
```

**Review by mustard-mh — COMMENTED** [review]
Left some comments

**Comment by mustard-mh** [comment]
Will go test it in Gitpod and preview env, since comments above are all not blocking

File: src/vs/gitpod/browser/gitpodInsightsAppender.ts (hunk 2 above)
Comment (jeanp413) [reply src/vs/gitpod/browser/gitpodInsightsAppender.ts:86]: Is different for the server and the browser, server has access to process.env, browser doesn't

**Comment by jeanp413** [comment]
@mustard-mh it works now :tada:

**Comment by mustard-mh** [comment]
session id will be redacted, so we need to make it trusted?

image

**Review by mustard-mh — APPROVED** [review]
Approve, but not merge if we want to make sessionId trusted https://github.com/gitpod-io/openvscode-server/pull/513#issuecomment-1695182526

**Comment by jeanp413** [comment]
> session id will be redacted, so we need to make it trusted? here?

I checked and the request is done with the original value, maybe it's just something visual from segment UI, if not we could rename the field

**Pull request merged**